### PR TITLE
OKD 4.4: run e2e test on machine-os-content promotion

### PR DESCRIPTION
**Pull request by vrutkovs**
Ensure new FCOS changes don't break OKD. This ensures pending machine-os-content update is being tested on AWS.

This also includes a fix for `vmtoolsd.service` symlink (fixes https://github.com/openshift/okd/issues/46)

Test results: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/6558/rehearse-6558-promote-release-openshift-okd-machine-os-content-e2e-aws-4.4/61

### CODE REVIEW

File: ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml [hunk 1]
```diff
@@ -558,10 +580,30 @@ periodics:
       - name: CLUSTER_TYPE
         value: aws
       - name: CONFIG_SPEC
-        valueFrom:
-          configMapKeyRef:
-            key: openshift-origin-release-4.4.yaml
-            name: ci-operator-4.4-configs
+        value: |
```
Comment (LorbusChris) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml]: why is this needed?

**Comment by LorbusChris** [comment]
Could you split out the vsphere changes into another PR for clarity? Let's make it work on AWS first

File: ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml (hunk 1 above)
Comment (LorbusChris) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml]: oh is it because it would otherwise run _all_ the tests, whereas we just to run the e2e test?

File: ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml [hunk 2]
```diff
@@ -557,10 +579,30 @@ periodics:
...
+          tag_specification:
+            name: "none" # don't import anything to avoid a race between 4.2 latest and the chosen version
```
Comment (smarterclayton) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml]: You shouldn't race anything.  I think this is structured wrong.  You shouldn't even have to create a release image.  What you need to do is run ci-operator twice - once before you do the builds to do the imports, then run your build to override the image, then run it again and it will generate the payload for you.

So your definition should look like a standard okd branch definition, minus the build jobs.  You want to run to `--target=[images]`, then you run your build and have it override `stable:machine-os-content`.

**Comment by vrutkovs** [comment]
Rehearses are borked?

/retest

File: ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml (hunk 2 above)
Comment (vrutkovs) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml]: So I tried that - and I can't make the script find the create namespace. It appears `namespace="ci-op-$( echo "${JOB_NAME}-${BUILD_ID}" | md5sum -t | cut -c 1-8 )"` doesn't work, cause `ci-tools` now includes a timestamp.

**Comment by vrutkovs** [comment]
flakes

/retest

**Comment by vrutkovs** [comment]
Marketplace flakes

/retest

File: ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml (hunk 1 above)
Comment (smarterclayton) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml]: You don't need to run all tests, that is what `--target TEST_NAME` does.

Comment (vrutkovs) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml]: A custom `CONFIG_SPEC` section is required to import machine-os-content from `fcos` (see `base_images` there) and `promotion` section (added `--promote` option to ci-operator)
